### ISSUE TITLE
Should transformations between spatial reference systems be supported?

### ISSUE THREAD

**Issue by FransKnibbe** [opengeospatial/ogc-geosparql#110]
A question that arose at the 2021-03-10 teleconference: Should GeoSPARQL have a function to allow transformation of geometries from one spatial reference system (SRS) to another? Such a function is available in many GIS software ([PostGIS for example](https://postgis.net/docs/ST_Transform.html)).

**Comment by FransKnibbe** [comment]
My thoughts: SRS transformation is a very useful function for many applications. But I think it does need a framework for defining spatial reference systems. A coordinate transformation function needs to be able to access the required SRS parameters. Assuming we prefer SRSs to be identified and accessed by IRI, an SRS transformation function would require an SRS ontology (see [issue 12](12)).

**Comment by situx** [comment]
I have created a draft pull request with some suggestions for this #108 and a proposal on whether we should also support transformations between literal formats. I looked at POSTGIS as an example.
Two common function signatures exist in POSTGIS from which we could infer:
geof:transform(geom: ogc:geomLiteral, from: xsd:anyURI, to: xsd:anyURI) ogc:geomLiteral
geof:transform(geom: ogc:geomLiteral, to: xsd:anyURI) ogc:geomLiteral
The first requires the source SRS to be stated, the second retrieves it from the literal (or possibly) another property in the background for GeoSPARQL 2.0.
In my opinion the second signature might be sufficient, the first signature might be interesting if we were to define geometry literals without a SRS attached.

I second the need for a SRS ontology, but for GeoSPARQL 1.1 we could stick to the definition of a SRS as an IRI as it is currently done in WKT and GML literals and rely on the triplestore implementation to provide their own database to resolve the SRS parameters.

**Comment by mathib** [comment]
@situx the second function is interesting and useful indeed for geometries with an unknown SRS!

**Comment by situx** [comment]
> @situx the second function is interesting and useful indeed for geometries with an unknown SRS!

You are right but I think using GeoSPARQL 1.0 the CRS cannot be undefined or unknown in all literal types as we established before.

So maybe one could add the second signature when it is possible to define those in the standard

**Comment by jabhay** [comment]
Adding a second signature to the standard sounds like something we'd like to do, but probably this function is not alone. A good candidate for version 1.2. I'll add it to the proposed features for that version.
